packages/puppeteer-core/src/node/ScreenRecorder.ts

```diff
@@ -183,21 +215,39 @@ export class ScreenRecorder extends PassThrough {
           // Sets the format
           ['-f', 'webm'],
           // Sets the quality. Lower the better.
-          ['-crf', `${CRF_VALUE}`],
+          ['-crf', `${quality}`],
           // Sets the quality and how efficient the compression will be.
           ['-deadline', 'realtime', '-cpu-used', '8'],
         ].flat();
       case 'gif':
+        fps = DEFAULT_FPS === fps ? 20 : 'fps=source_fps';
+        if (!Number.isInteger(loop) || loop === 0) loop = ~~!loop * -1;
+        if (!Math.sign(delay)) delay /= 10; // ms to cs
         return [
           // Sets the frame rate and uses a custom palette generated from the
           // input.
           [
             '-vf',
-            'fps=5,split[s0][s1];[s0]palettegen=stats_mode=diff[p];[s1][p]paletteuse',
+            `fps=${fps},split[s0][s1];[s0]palettegen=stats_mode=diff:max_colors=${colors}[p];[s1][p]paletteuse=dither=bayer`,
           ],
+          // Sets the number of times to loop playback.
+          ['-loop', `${loop}`],
+          // Sets the delay between iterations of a loop.
+          ['-final_delay', `${delay}`],
           // Sets the format
           ['-f', 'gif'],
         ].flat();
+      case 'mp4':
+        return [
+          // Sets the codec to use.
+          ['-c:v', 'libx264'],
+          // Sets the format
+          ['-f', 'mp4'],
+          // Sets the quality. Lower the better.
+          ['-crf', `${quality}`],
+          // Sets the quality and how efficient the compression will be.
+          ['-deadline', 'realtime', '-cpu-used', '8'],
```

Comment:
What will happen if the Computer has less then 8 CPU cores?

Reply:
@Lightning00Blade Presumably, it will explode! Seems to be another oversight in the existing code. Fixed now…
~~~ js
import os from 'node:os'
// …
, '-cpu-used', `${os.cpus().length / 2}`],
~~~